src/fsharp/vs/Exprs.fs

```diff
@@ -752,6 +679,190 @@ module FSharpExprConvert =
             let envinner = env.BindVal v
             Some(vR,rhsR),envinner
 
+    and ConvILCall cenv env (isNewObj, valUseFlags, ilMethRef, enclTypeArgs, methTypeArgs, callArgs, m) =
+        let isNewObj = (isNewObj || (match valUseFlags with CtorValUsedAsSuperInit | CtorValUsedAsSelfInit -> true | _ -> false))
+        let methName = ilMethRef.Name
+        let isPropGet = methName.StartsWith("get_",System.StringComparison.Ordinal)
+        let isPropSet = methName.StartsWith("set_",System.StringComparison.Ordinal)
+        let isProp = isPropGet || isPropSet
+        
+        let tcref, subClass = 
+            // this does not matter currently, type checking fails to resolve it when a TP references a union case subclass
+            try
+                // if the type is an union case class, lookup will fail 
+                Import.ImportILTypeRef cenv.amap m ilMethRef.EnclosingTypeRef, None
+            with _ ->
+                let e = ilMethRef.EnclosingTypeRef
+                let parent = ILTypeRef.Create(e.Scope, e.Enclosing.Tail, e.Enclosing.Head)
+                Import.ImportILTypeRef cenv.amap m parent, Some e.Name
+                
+        let enclosingType = generalizedTyconRef tcref
+        
+        let makeCall minfo =
+            ConvObjectModelCallLinear cenv env (isNewObj, minfo, enclTypeArgs, methTypeArgs, callArgs) id   
+
+        let makeFSCall isMember (vr: ValRef) =
```

Comment:
Please extract ``makeFSCall`` and ``makeFSExpr`` out to helper functions in a module at the top of the file, independent of the big ``let rec``.  Along the lines of ``bindILMethodRefToSymbol`` in the previous code.  I'd like to see that they are `///` comments and be independent of the rest of the code.